### DISCUSSION TITLE
Pie chart order

### DISCUSSION THREAD

**Discussion by hoclun-rigsep** [vega/vega-lite#7903] [Q&A] [answered]
Here's the pie chart example:

```vega-lite
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "description": "A simple pie chart with embedded data.",
  "data": {
    "values": [
      {"category": 1, "value": 4},
      {"category": 2, "value": 6},
      {"category": 3, "value": 10},
      {"category": 4, "value": 3},
      {"category": 5, "value": 7},
      {"category": 6, "value": 8}
    ]
  },
  "mark": "arc",
  "encoding": {
    "theta": {"field": "value", "type": "quantitative"},
    "color": {"field": "category", "type": "nominal"}
  }
}
```

How can I order the resulting sectors of the pie chart by value? I tried, among other things, putting `"sort": "descending"` in the theta encoding but this did not work. I also converted the example to a bar chart and it was straightforward to order the bars in this way. Is this a known trouble spot or have I just had some kind of brain lapse?

**Comment by jakevdp** [comment, marked as answer]
If you want the color scale to be ordered by value, you can use `"sort": {"field": "value"}` within the color encoding. If you want the placement of the pie segments to be ordered by value, you can use `"order": {"field": "value"}` within the encoding. Putting both together:
```json
{
  "data": {
    "values": [
      {"category": 1, "value": 4},
      {"category": 2, "value": 6},
      {"category": 3, "value": 10},
      {"category": 4, "value": 3},
      {"category": 5, "value": 7},
      {"category": 6, "value": 8}
    ]
  },
  "mark": "arc",
  "encoding": {
    "theta": {"field": "value", "type": "quantitative"},
    "color": {"field": "category", "type": "nominal", "sort": {"field": "value"}},
    "order": {"field": "value"}
  }
}
```
